docs/TuxReact.md

```diff
@@ -0,0 +1,449 @@
+
+***
+
+#### <a id="React-Components"></a>3) React Components [#](#React-Components)
+`React` components are defined via the `React.createClass` method. Pass in an object to this method, and `React` will return a class which you can then use to render components built from that class into the DOM:
+
+```
+    var React = require('react');
+    var MyComponent = React.createClass({
+      render: function () {
+        return (
+          <h1>Hello World!</h1>
+        );
+      }
+    });
+```
+
+***
+
+#### <a id="React-Components-the-Render-Method"></a>3.1) the Render Method [#](#React-Components-the-Render-Method)
+All `React` classes must have a `render` method. The `render` method must return a single `React` component (either a basic `React` html component for rendering onto the DOM (`h1`, `div`, `p`, etc) or a `React` composite class (any custom build `React` component, for example `MyComponent` defined above is a `React` composite class).
+
+It might seem like returning only a single `React` component is a bit limiting. However, you can pass in child components to whatever `React` component you render. Will get more into what that actually means in the next section, but in laymens terms it means you can `render` however many components you want:
+
+```
+    var React = require('react');
+    var MyComplicatedComponent = React.createClass({
+      var someArrayOfComponents = [<ComponentA key={0} />, <ComponentB key={1} />];
+      render: function () {
+        return (
+          <div>
+            <MyComponent className='custom-class-name' />
+            <h1>Let's render something {thisComponentIsAwesome ? 'Awesome' : 'Lame'}</h1>
+            { someArrayOfComponents }
+          </div>
+        );
+      }
+    });
+```
+
+**NOTES**
+1. We are wrapping `someArrayOfComponents` and our ternary expression in `{}`. That's because, in order to use Javascript variables and expressions in our JSX we need to wrap them in `{}` and we need to use ternary expressions if we want to write if else statements in our JSX. You can learn more about if-else in JSX [here](http://facebook.github.io/react/tips/if-else-in-JSX.html).
+2. We are passing in `key` properties to our components in our array. Whenever we render an array of components we need to pass in a unique key to each one, which `React` will use to determine which components in the array have been modified and which have been deleted/added when the array updates. You can't access the `key` property from within the component itself.
+2. There are several restrictions regarding what Javascript we can write in our JSX and you can read more about that [here](http://facebook.github.io/react/docs/jsx-in-depth.html#javascript-expressions).
+3. We are passing in a property `className`. This, is because `class` is a reserved word in Javascript. See [here](http://facebook.github.io/react/docs/jsx-in-depth.html#html-tags-vs.-react-components) for some other differences between JSX and html
```

Comment:
Change instances of html that are not file extensions to HTML. Apply throughout document.